### DISCUSSION TITLE
Instructions on how to set up a minimal VS studio project

### DISCUSSION THREAD

**Discussion by meghakalia** [PlusToolkit/PlusLib#969] [Q&A]
Hi, 
I am trying to use the volume recontsruction class. The lib and dlls are in common foldres but includes are scattered all across the project. Do you have any information on how to get necessary includes in a one folder so that project runs. 

Currently, I am manually pointing to the necesary includes, which is quite a painful process. 

Will apprecite any pointers.

**Comment by Sunderlandkyl** [comment]
Are you following the instructions from [PlusBuild](https://github.com/PlusToolkit/PlusBuild/blob/master/README.md)?

Compiling Plus through PlusBuild is the easiest way to manage all of the different projects.

**Reply by meghakalia** [reply]
Hi, 

Thanks for replying. Yes build is successful both for debug and release. lib and dlls are in separate folders for the project. Where are the includes though? In VS2017 project I have #include <src/PlusVolumeReconstruction/vtkPlusVolumeReconstructor.h>. I have path to this particular include in Additional include directories. However, that gives me tons of errors that the file not found. Basically this file needs to see other dependent includes and definitions. How to resolve this ? I am thinking of getting ALL the includes in one folder and then pointing to that folder. Happy to hear other better solutions though.

**Reply by adamrankin** [reply]
Are you using the CMake build system? Plus is a highly CMake-based project and these issues are easily resolved by using CMake.

**Comment by adamrankin** [comment]
Manually set the `CMAKE_INSTALL_PREFIX` for the PlusLib project, then build the INSTALL project in VS. The folder you choose will have a subfolder called `include\PlusLib-2.9` that contains all necessary includes.

**Reply by meghakalia** [reply]
Hi, 

Thanks for your reply. I tried your solution. It did not solve the problem though. Or rather it seems like a partial solution. 

include\PlusLib-2.9 has two file, namely: 
vtkPlusVolumeReconstructionExport.h
vtkPlusVolumeReconstructor.h

However, 
in vtkPlusVolumeReconstructor.h there are 4 includes

#include "PlusConfigure.h"
#include "vtkPlusVolumeReconstructionExport.h"

// IGSIO includes
#include <igsioCommon.h>
#include <vtkIGSIOVolumeReconstructor.h>

The code still complains that it did not find the other includes which are #include "PlusConfigure.h" and // IGSIO includes. 

Do you have any suggestions ? 
Thanks in advance

**Reply by adamrankin** [reply]
You should have 50+ files in that folder. Did you build Plus via the PlusBuild solution? For the igsio files, you can do the same procedure by setting the `CMAKE_INSTALL_PREFIX` for the `IGSIO\inner-build` project and that will produce a `include\IGSIO-1.0` subfolder.

**Reply by meghakalia** [reply]
After modifying the cmakefile for that project. I built only that specific project: vtkPlusVolumeReconstruction and Built the INSTALL as well. Let me try to build the PlusLib from PlusBuild sln again and update. 
Thanks

**Reply by meghakalia** [reply]
CMAKE_INSTALL_PREFIX for IGSIO in inner-build in the install directory is not creating any include folder. Only lib and bin are created. Is there a specific place where I should set CMAKE_INSTALL_PREFIX . Thanks in advance.
